src/main/webapp/app/shared/metis/posting-markdown-editor/posting-markdown-editor.component.ts

```diff
@@ -76,6 +79,9 @@ export class PostingMarkdownEditorComponent implements OnInit, ControlValueAcces
         private courseManagementService: CourseManagementService,
         private lectureService: LectureService,
         private channelService: ChannelService,
+        public viewContainerRef: ViewContainerRef,
+        private overlay: Overlay,
+        private positionBuilder: OverlayPositionBuilder,
```

Comment:
I think you could use the new way of injecting here:
`private overlay = inject(Overlay);` etc.